opentelemetry-sdk/tests/metrics/integration_test/test_histogram_export.py

```diff
@@ -0,0 +1,85 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import TestCase
+
+from opentelemetry.metrics import set_meter_provider
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import InMemoryMetricReader
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+
+in_memory_metric_reader = InMemoryMetricReader()
+
+provider = MeterProvider(
+    resource=Resource.create({SERVICE_NAME: "otel-test"}),
+    metric_readers=[in_memory_metric_reader],
+)
+set_meter_provider(provider)
+
+meter = provider.get_meter("my-meter")
+
+histogram = meter.create_histogram("my_histogram")
+counter = meter.create_counter("my_counter")
+
+
+class TestHistogramExport(TestCase):
+    def test_histogram_counter_collection(self):
```

Comment:
I'm confused how this test is related to the change in this PR, maybe it can be removed now based on previous discussion

Reply:
I don't think this PR fixes the issue, I outlined the real problem in https://github.com/open-telemetry/opentelemetry-python/issues/3277#issuecomment-1684545842

Can you move this test to a separate PR.

Reply:
#3277 name is: "Histograms are exported empty if no observations are recorded during a reading interval"

With this PR histograms are no longer exported empty if no observations are recorded during a reading interval, thus, this PR fixes #3277 and to show it does I added this test case.

I understand that while reviewing #3277 you found another, more serious issue. I opened #3407 to track this other, more serious issue. I'll add a fix for #3407 in a nother PR, and when that happens I'll add a new test case specific for #3407, will update the test case added here and remove [this](https://github.com/open-telemetry/opentelemetry-python/pull/3335/files#diff-82de071cc175f58c01255554362a1f0b33aa1d55a2e9a65e0221158018b866c3R68) line.

The purpose of this specific test case is just to show that the behavior reported in #3277 no longer happens.